lib/devise/rails/routes.rb

```diff
@@ -415,7 +415,12 @@ def devise_omniauth_callback(mapping, controllers) #:nodoc:
 ERROR
         end
 
-        path, @scope[:path] = @scope[:path], nil
+        path = @scope[:path]
```

Comment:
Using `@scope.frame` seems to fix #3701, but I wonder if we should try something with `@scope.new` and `@scope.parent` from @tenderlove patch from that thread.